### PR TITLE
refactor: update webhooks HOF to Webhooks class

### PR DESCRIPTION
**Pull request by karrui**
Part one of the refactor, meant to keep the PR size manageable.

Related to #31 

The current chain is:
master -> **webhooks (this)**

### CODE REVIEW

File: src/webhooks.ts [hunk 1]
```diff
@@ -1,109 +1,74 @@
...
       s: submissionId,
       f: formId,
-    } = parseSignatureHeader(header)
-    const epoch = Number(t)
+    } = signatureHeader
 
     if (!epoch || !signature || !submissionId || !formId) {
-      throw new Error('X-FormSG-Signature header is invalid')
+      throw new WebhookAuthenticateError('X-FormSG-Signature header is invalid')
     }
```
Comment (arshadali172) [inline src/webhooks.ts:42]: I think these checks should be part of `isSignatureHeaderValid()` no?

File: src/webhooks.ts [hunk 2]
```diff
@@ -113,54 +78,42 @@ function generateSignature(webhookSecretKey: string) {
     submissionId: Object
     formId: string
     epoch: number
-  }) {
+  }) => {
+    if (!this.secretKey) {
```
Comment (arshadali172) [inline src/webhooks.ts:82]: Can we move this check to the constructor?

File: src/webhooks.ts (hunk 1 above)
Comment (karrui) [reply src/webhooks.ts]: Yea, agreed. Moved into validation function.

File: src/webhooks.ts (hunk 2 above)
Comment (karrui) [reply src/webhooks.ts:78]: As discussed, left here for flexibility on function calls. The `authenticate` function does not need a secret key to be called.

**Pull request merged**